internal/pkg/application/models.go

```diff
@@ -69,3 +69,15 @@ func MapToModel(d database.Device) Device {
 		Tenant:       t,
 	}
 }
+
+type StatusMessage struct {
+	DeviceID  string  `json:"deviceID"`
+	Error     *string `json:"error,omitempty"`
+	Status    Status  `json:"status"`
+	Timestamp string  `json:"timestamp"`
+}
+
+type Status struct {
+	Code     int      `json:"statusCode"`
+	Messages []string `json:"statusMessages,omitempty"`
```

Comment:
Funkar omitempty på en slice? Frågar åt en vän.

Reply:
The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.
https://pkg.go.dev/encoding/json